src/test/java/BaseTest.java

```diff
@@ -1,11 +1,108 @@
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Parameters;
 
-public class BaseTest {
+import java.time.Duration;
+import java.util.UUID;
 
+public class BaseTest {
+    public WebDriver driver = null;
+    public WebDriverWait wait;
+    public static Actions actions;
+    public String url = "https://qa.koel.app/";
     @BeforeSuite
-    static void setupClass() {
+    void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
-}
\ No newline at end of file
+    @BeforeMethod
+    @Parameters({"BaseURL"})
+    public void launchBrowser(String baseURL) {
+
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--remote-allow-origins=*");
+
+        driver = new ChromeDriver(options);
+        // implicit wait
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+        //explicit wait
+        wait = new WebDriverWait(driver,Duration.ofSeconds(10));
+        driver.manage().window().maximize();
+
+        navigateToPage(baseURL);
+        actions =new Actions(driver);
+    }
+    @AfterMethod
+    public void closeBrowser() {
+        driver.quit();
+    }
+
+
+    public void navigateToPage(String url) {
+        driver.get(url);
+    }
+
+    public void provideEmail(String email) {
+        //WebElement emailField = driver.findElement(By.cssSelector("input[type='email']"));
+        WebElement emailField = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("input[type='email']")));
+        emailField.clear();
+        emailField.sendKeys(email);
+    }
+
+    public void providePassword(String password) {
+        //WebElement passwordField = driver.findElement(By.cssSelector("input[type='password']"));
+        WebElement passwordField =wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("input[type='password']")));
+        passwordField.clear();
+        passwordField.sendKeys(password);
+    }
+
+    public void clickSubmit() {
+        //WebElement submit = driver.findElement(By.cssSelector("button[type='submit']"));
+        WebElement submit =wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("button[type='submit']")));
+        submit.click();
+    }
```

Comment:
Page Object Model (POM): recommended to associate the methods with the corresponding page class where they are relevant. In the POM, each page or section of a web application is represented by a separate class. These classes encapsulate the elements and actions specific to that page or section.

Move these methods to the corresponding class according to the POM.